api/src/pcapi/static/backoffice/js/pc-table-multi-select.js

```diff
@@ -1,69 +1,94 @@
+/**
+ * TODO: add basic documentation
+ */
+class TableMultiSelect {
+    rowsIds = []
+    selectedRowsIds = []
+    $container = undefined
 
-var idValidations = [] //rows selected
+    constructor($container) {
+        this.$container = $container
+        this.initialize()
+        this.bindEvents()
+    }
 
-function popAtIndex(array, element) { //Delete from array
-    const index = array.indexOf(element);
-    if (index > -1) {
-        array.splice(index, 1)
+    get $checkboxes() {
+        return this.$container.querySelectorAll("input[name^='pctms-check-']")
     }
-}
-function selectAll() {
-    allCheckboxes.forEach(el => {
-        el.checked = true
-        if (el.id !== allValidationsId) {
-            idValidations.push(el.id)
-        }
-    })
-}
 
-function unselectAll() {
-    allCheckboxes.forEach(el => {
-        el.checked = false
-    })
-}
+    get $batchButtons() {
+        return this.$container.querySelectorAll('.pctms-batch-button-group > button')
+    }
 
-function disableToolbar(disable) {
-    Array.from(document.getElementById("batch-buttons").children).forEach(el => {
-        el.disabled = disable
-    })
-}
+    get $checkboxAll() {
+        return this.$container.querySelector('.pctms-check-all')
+    }
 
-if (idValidations.length === 0) {
-    disableToolbar(true)
-}
+    initialize = () => {
+        if (this.selectedRowsIds.length === 0) {
+            this.disableToolbar(true)
+        }
+        this.$checkboxes.forEach(($el) => {
+            if ($el.id !== this.selectedRowsIds) {
+                this.rowsIds.push($el.id)
+            }
+        })
+    }
 
+    bindEvents = () => {
+        this.$checkboxes.forEach(($checkbox) => {
+            $checkbox.addEventListener("click", (event) => {
+                if ($checkbox.checked) {
+                    this.disableToolbar(false)
+                    if (event.currentTarget.name === this.$checkboxAll.name) {
+                        this.selectedRowsIds = []
+                        this.selectAll()
+                    } else {
+                        this.selectedRowsIds.push($checkbox.dataset.id)
+                    }
+                } else {
+                    if (event.currentTarget.name === this.$checkboxAll.name) {
+                        this.unselectAll()
+                        this.selectedRowsIds = []
+                        this.disableToolbar(true)
+                    } else {
+                        this.popAtIndex(this.selectedRowsIds, $checkbox.dataset.id)
+                        if (this.selectedRowsIds.length === 0) {
+                            this.disableToolbar(true)
+                        }
+                    }
+                }
 
-allCheckboxes.forEach((el) => {
-    //Initialize the array of id of all rows
-    if (el.id !== allValidationsId) {
-        allIds.push(el.id)
+                // Manage The indeterminate state of select-all checkbox
```

Comment:
`Manage the indeterminate state of select-all checkboxes/ of the select-all checkbox`